### PR TITLE
Browser UI for browsing area maps in database

### PR DESCRIPTION
**Pull request by rhietala**

### CODE REVIEW

File: mapbrowser/templates/base.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      <header>
+        {% block header %}{% endblock %}
+      </header>
+      {% for message in get_flashed_messages() %}
+        <div class="flash">{{ message }}</div>
+      {% endfor %}
+      {% block content %}{% endblock %}
+    </section>
+  </body>
+</html>
```
Comment (lotheac) [inline mapbrowser/templates/base.html:18]: add newline

File: mapbrowser/templates/map.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      labelSize: 'fixed',
+      labelSizeRatio: 0.1,
+      defaultNodeType: 'longdesc',
+      font: 'monospace',
+      minArrowSize: 8,
+    },
+    });
+    sigma.parsers.json('/data/{{ area }}', s, function() { CustomShapes.init(s); s.refresh(); });
+  </script>
+{% endblock %}
```
Comment (lotheac) [inline mapbrowser/templates/map.html:90]: add newline

File: mapbrowser/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+__pycache__/
```
Comment (lotheac) [inline mapbrowser/.gitignore:1]: add newline

**Review by lotheac — CHANGES_REQUESTED** [review]
So, some general feedback here:

area_to_json.py is quite slow, which is why my initial PoCs pregenerated the json for sigma to be consumed; you've essentially copied that logic here, which makes page loads slow, but that is fine for now and is a good start.

On a higher architectural level, the splitting into subgraphs  and their respective positioning should be controllable by the end user (since it's quite difficult to lay out an arbitrarily complicated map programmatically). So this PoC is fine, but we need to eventually extend the db format to store information about at least which subgraph each node belongs to, and the positioning of the subgraphs, and room notes. Maybe also individual node positioning if necessary.

So if we can store some of the layout/positioning data in the db, then the common case for page loads should be fast -- we would only need to run the expensive/horribly-ugly logic for initial positioning if there are some nodes in the graph that don't have subgraph/positioning data.

Other than that just some nitpicks about newlines.

**Comment by rhietala** [comment]
Instead of merging the mapbrowser web app here, let's continue development in its own repo https://github.com/bat-boys/bcproxy-mapbrowser/

**Pull request closed without merging**